op-e2e/e2eutils/retryproxy/proxy.go

```diff
@@ -0,0 +1,160 @@
+package retryproxy
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var copyHeaders = []string{
+	"Content-Type",
+}
+
+type RetryProxy struct {
+	lgr        log.Logger
+	upstream   string
+	client     *http.Client
+	strategy   retry.Strategy
+	maxRetries int
+	srv        *http.Server
+	listenPort int
+}
+
+type Option func(*RetryProxy)
+
+func New(lgr log.Logger, upstream string, opts ...Option) *RetryProxy {
```

Comment:
I'm wondering how useful the `opts` API is, when it only has access to a struct without any exported members? I don't think this lets me, for example, modify the max retries in an `Option` defined in a different package?